ros2topic/test/test_use_sim_time.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2020 Sony Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import functools
+import sys
+import unittest
+
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess
+
+import launch_testing
+import launch_testing.actions
+import launch_testing.asserts
+import launch_testing.markers
+import launch_testing.tools
+import launch_testing_ros.tools
+
+import pytest
+
+import rclpy
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.utilities import get_rmw_implementation_identifier
+
+from std_msgs.msg import String
+from rosgraph_msgs.msg import Clock
+
+
+# Skip cli tests on Windows while they exhibit pathological behavior
+# https://github.com/ros2/build_farmer/issues/248
+if sys.platform.startswith('win'):
+    pytest.skip(
+            'CLI tests can block for a pathological amount of time on Windows.',
+            allow_module_level=True)
+
+
+TEST_NODE = 'cli_use_sim_time_test_node'
+TEST_NAMESPACE = 'cli_use_sim_time'
+
+
+@pytest.mark.rostest
+@launch_testing.markers.keep_alive
+def generate_test_description():
+    return LaunchDescription([
+        # Always restart daemon to isolate tests.
+        ExecuteProcess(
+            cmd=['ros2', 'daemon', 'stop'],
+            name='daemon-stop',
+            on_exit=[
+                ExecuteProcess(
+                    cmd=['ros2', 'daemon', 'start'],
+                    name='daemon-start',
+                    on_exit=[
+                        launch_testing.actions.ReadyToTest()
+                    ],
+                )
+            ]
+        )
+    ])
+
+
+class TestROS2TopicUseSimTime(unittest.TestCase):
+
+    def timer_callback(self):
+        msg = Clock()
+        msg.clock.sec = self.clock_sec
+        self.publisher.publish(msg)
+        self.clock_sec += 1
+
+    def setUp(self):
+        self.context = rclpy.context.Context()
+        rclpy.init(context=self.context)
+        self.node = rclpy.create_node(
+            TEST_NODE, namespace=TEST_NAMESPACE, context=self.context
+        )
+        self.publisher = self.node.create_publisher(Clock, '/clock', 10)
+        self.executor = SingleThreadedExecutor(context=self.context)
+        self.executor.add_node(self.node)
+
+    def tearDown(self):
+        self.node.destroy_node()
+        rclpy.shutdown(context=self.context)
+
+    @launch_testing.markers.retry_on_failure(times=5)
```

Comment:
The test method of below 3 test cases need to be updated based on our offline discussion.

Reply:
Updated.